### ISSUE TITLE
Target encoding & early stopping

### ISSUE THREAD

**Issue by alizat** [alizat/Life-with-DL#6]
Two more tricks for the [irrigation](https://github.com/alizat/Life-with-DL/tree/main/B_Fundies/01_irrigation) project:
* Target encoding to be used instead of one-hot encoding, should improve prediction performance
* Early stopping to avoid overtraining: try a smaller learning rate and much more epochs, let the model decide when to stop training

**Comment by alizat** [comment]
Target encoding not possible due to the target variable having more than 2 possible values!
